app/backend/src/clj/teet/authorization/authorization_core.clj

```diff
@@ -0,0 +1,48 @@
+(ns teet.authorization.authorization-core
+  "Authorize user actions based on their role in contracts, or global roles given through admin interface"
+  (:require [clojure.java.io :as io]
+            [teet.authorization.authorization-db :as authorization-db]
+            [clojure.set :as set]
+            [teet.user.user-model :as user-model]
+            [teet.user.user-db :as user-db]))
+
+(defonce authorization-matrix
+         (delay
+           (some-> "contract-authorization.edn"
+                   io/resource
+                   slurp
+                   read-string)))
+
+(defn project-read-access?
+  [db user project-id]
+  (let [user-ref (user-model/user-ref user)
+        user-global-roles (user-db/users-valid-global-permissions db user-ref)
+        user-roles-for-project (authorization-db/user-roles-for-project db user-ref project-id)]
+    (-> (set/union user-global-roles user-roles-for-project)
+        seq
+        boolean)))
+
+(defn authorized-for-action?
+  "Check rights either based on the target or company
+  Given either company or a target find if the user has the right "
+  [{:keys [db user action target company]}]
+  {:pre [(or target company)
+         (not (and target company))]}
+  (let [authorized-roles (action @authorization-matrix)
```

Comment:
should you assert that action is given and exists in the authorization matrix?
now giving `nil` as action will throw NPE

Reply:
also it seems `db` is required, might as well pre check that is given